test_runner/fixtures/pageserver/allowed_errors.py

```diff
@@ -111,6 +111,10 @@ def scan_pageserver_log_for_errors(
     ".*stalling layer flushes for compaction backpressure.*",
     ".*layer roll waiting for flush due to compaction backpressure.*",
     ".*BatchSpanProcessor.*",
+    # Tests run in dev mode
+    ".*Starting in dev mode.*",
+    ".*Starting in dev mode - authentication security checks are disabled.*",
```

Comment:
`.*Starting in dev mode.*` regexp already matches all other messages you added. Delete excessive log messages you added